apiserver/facades/client/usermanager/usermanager.go

```diff
@@ -383,3 +383,43 @@ func (api *UserManagerAPI) setPassword(arg params.EntityPassword) error {
 	}
 	return nil
 }
+
+// ResetPassword resets password for supplied users by
+// invalidating current passwords (if any) and generating
+// new random secret keys which will be returned.
+func (api *UserManagerAPI) ResetPassword(args params.Entities) (params.AddUserResults, error) {
+	var result params.AddUserResults
+
+	if err := api.check.ChangeAllowed(); err != nil {
+		return result, errors.Trace(err)
+	}
+
+	if len(args.Entities) == 0 {
+		return result, nil
+	}
+
+	isSuperUser, err := api.hasControllerAdminAccess()
```

Comment:
I wonder if we want to support people directly requesting to reset their own passwords. (You happen to still be authed, but you've forgotten your password, so you reset it so that you can set it to something you can remember)

If we did, then we'd need a "isSuperUser || api.User == entity.User" sort of check. and the ErrPerm here would certainly need to be per entity, rather than for the overall request.

Thoughts?